packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx

```diff
@@ -60,15 +61,21 @@ type SavedPipelinesProps = {
   savedPipelines: StoredPipeline[];
   onOpenPipeline: (pipelineData: StoredPipeline) => void;
   onDeletePipeline: (pipelineId: string) => void;
+  onMount: () => void;
 };
 
 export const SavedPipelines = ({
   namespace,
   savedPipelines,
   onOpenPipeline,
   onDeletePipeline,
+  onMount,
 }: SavedPipelinesProps) => {
   const darkMode = useDarkMode();
+  const onMountRef = useRef(onMount);
+  useEffect(() => {
+    onMountRef.current();
+  }, []);
```

Comment:
Is this to avoid multiple calls to `getSavedPipelines`?

Reply:
Yes, this will be checked in the state against isLoaded prop, but even taking this into account I want some sort of normal interface for this on the component, not just passing the method directly and calling it inside the render, side-effects like that should be avoided